### DISCUSSION TITLE
Add vercel analytics in `@astrojs/vercel`

### DISCUSSION THREAD

**Discussion by jsun969** [withastro/roadmap#467] [Proposal]
# Summary

Add vercel analytics in `@astrojs/vercel`

# Background & Motivation

I am using vercel to deploy my astro app. I but I have to manually add **web vituals** and **audiences** scripts.

![image](https://user-images.githubusercontent.com/29330847/216648194-e0478f3f-5068-4917-aadd-6c791d7c0aa7.png)

# Goals

Add optons in `@astrojs/vercel` to enable web vituals and audiences for vercel analytics.

# Example

I have done it manually in my project. Hope to be able to integrate.

https://github.com/jsun969/jsundotlol/blob/22fb7fe76b2d161960cfc7bd9f0e4ae0d62dea97/src/scripts/vercelAnalytics.ts
https://github.com/jsun969/jsundotlol/blob/22fb7fe76b2d161960cfc7bd9f0e4ae0d62dea97/src/scripts/vitals.ts

**Comment by natemoo-re** [comment]
This seems pretty reasonable! Would you be willing to create a PR for `@astrojs/vercel` that adds an `analytics: true` option that would inject these scripts for the user?

**Reply by jsun969** [reply]
For sure🙌

**Reply by jsun969** [reply]
Sry, I don't really understand the code of the vercel adapter and don't know how to join the file at build time. I wrote a blog post about it. Hope this helps the maintainer.

<https://jsun.lol/blog/enable-vercel-analytics-in-astro>

**Reply by delucis** [reply]
The `injectScript` API should work here, using `'page'` as its first argument ([docs](https://docs.astro.build/en/reference/integrations-reference/#injectscript-option)).

I’ve also previously set up a small component to help improve WebVitals tracking with generic paths to group data: https://github.com/delucis/astro-badge/blob/latest/src/components/TrackVitals.astro

**Reply by jsun969** [reply]
> The `injectScript` API should work here, using `'page'` as its first argument ([docs](https://docs.astro.build/en/reference/integrations-reference/#injectscript-option)).
> 
> I’ve also previously set up a small component to help improve WebVitals tracking with generic paths to group data: https://github.com/delucis/astro-badge/blob/latest/src/components/TrackVitals.astro

Thank you very much! I have created a PR for it: https://github.com/withastro/astro/pull/6148

**Reply by delucis** [reply]
Thanks so much @jsun969! Left a review there 🙌
